src/mower_logic/src/mower_logic/behaviors/DockingBehavior.cpp

```diff
@@ -203,8 +203,8 @@ Behavior *DockingBehavior::execute() {
         return &IdleBehavior::INSTANCE;
     }
 
-    // Reset retryCount
-    reset();
+    // Reset retryCount here would cause an infinite loop
```

Comment:
@ClemensElflein @bodop Are you sure about this? If retry_count is 0 and the dummy docking approach point is not reachable then should never get here.

I don't think the regular docking behaviour logic should be changed to accommodate a special case such as this. Would be better to add a new "disable_docking" param and just immediately return IdleBehavior.

Reply:
As I see it, the change does not have anything to do with the PR, it's a general docking loop issue, if `reset()` is called in line 207, then `retryCount` is always 1 in line 222, so for non-zero retries, the thing would always retry.

Reply:
Usually I'd agree with the "lets not add some weird behavior for a specific use case"-statement, but making the parameter value '0' valid seems ok to me.

Reply:
ok but changing the retry logic for dock_straight? This has changed the number of dock_straight attempts from config.docking_retry_count to (config.docking_retry_count - approach_attempts).

Don't understand how this can cause an infinite loop if docking_retry_count is 0. It would try dock_straight once and fail to Idle.

Reply:
ah, missed the prev comment. Ok, so it's to stop a Docking -> Undocking -> Docking loop when retry count > 0

Reply:
Just replaces one issue with a different issue though. Now a bad approach can disable dock_straight retries. Maybe should count approach and dock_stright retries separately?